Electronique/main.py

```diff
@@ -0,0 +1,60 @@
+from gpiozero import MotionSensor
+from picamera2 import Picamera2
+import time
+import cv2
+import os
+
+# Configuration de la caméra
+camera = Picamera2()
+
+# Configuration des capteurs de mouvement
+pir_out = MotionSensor(4)  # Capteur OUT
+pir_in = MotionSensor(27)  # Capteur IN
+
+try:
+    print("Attente du mouvement. Appuyez sur Ctrl+C pour quitter.")
+    time.sleep(2)  # Attend que le capteur s'initialise
+
+    while True:
+        if pir_out.motion_detected or pir_in.motion_detected:
+            print("Mouvement détecté! Enregistrement de la vidéo.")
+
+            # Détermine quel capteur a détecté le mouvement en premier
+            sensor_trigger = "in" if pir_in.motion_detected else "out"
+
+            # Récupère la date et l'heure pour le nom de fichier
+            timestamp = time.strftime("%d-%m-%y-%H-%M-%S", time.localtime())
+            filename = f"{sensor_trigger}-{timestamp}.mp4"
+            video_path = f"/home/cico/Videos/{filename}"
+
+            # Enregistre la vidéo pendant 10 secondes
+            camera.start_and_record_video(video_path, duration=10)
+            print(f"Vidéo enregistrée : {filename}")
+            
+            # Ouvre la vidéo
+            video_path = filename
+            cam = cv2.VideoCapture(video_path, cv2.CAP_FFMPEG)
+
+            frame_count = int(cam.get(cv2.CAP_PROP_FRAME_COUNT))
+            start_frame = frame_count - 4 if sensor_trigger == "in" else 0
```

Comment:
Question 5 : Elle sert à quoi cette ligne ? J'ai pas très bien compris le but de start_frame...

Reply:
Cette variable sert à choisir le moment où les 4 frames de la vidéo vont être enregistrées.
frame_count vaut le nombre total de frame de la vidéo enregistrée, si c'est une entrée "sensor_trigger" vaudra "in" et donc on choisit de prendre les 4 dernières frames le temps que le chat entre pour le voir sinon ça sera une sortie et dans ce cas la variable (start_frame) vaut 0 et on enregistre les frames dès le début de la vidéo pour ne pas rater le chat car il sort.